File: 하루에 한개씩 문제 풀기/Go/Programmers/Level 0/무작위로 K개의 수 뽑기.go
func solution(arr []int, k int) []int {
	uniqueNums := make(map[int]bool)
	result := []int{}

	for _, num := range arr {
		if !uniqueNums[num] {
			uniqueNums[num] = true
			result = append(result, num)
		}

		if len(result) == k {
			break
		}
	}

	for len(result) < k {
		result = append(result, -1)
	}

	return result
}

File: 하루에 한개씩 문제 풀기/Python/Programmers/모든 문제/Level 0/무작위로 k개의 수 뽑기.py
def solution(arr, k):
    unique_nums = set()
    result = []

    for num in arr:
        if num not in unique_nums:
            unique_nums.add(num)
            result.append(num)

        if len(result) == k:
            break

    while len(result) < k:
        result.append(-1)

    return result
